src/Symfony/Component/Mailer/Transport/Smtp/EsmtpTransport.php

```diff
@@ -54,6 +54,13 @@ public function __construct(string $host = 'localhost', int $port = 0, bool $tls
         }
         if (!$tls) {
             $stream->disableTls();
+        } elseif (!$verifyPeer) {
+            $stream->setStreamOptions(array_merge_recursive($stream->getStreamOptions(), [
+                'ssl' => [
+                    'verify_peer' => false,
```

Comment:
Don't you think adding `'allow_self_signed' => true,` would be a good idea to explicitly allow self-signed certificates?

Reply:
@nicolas-grekas what do you think about this?

Reply:
`verify_peer` is a pretty common name for this kind of setting. Also, the name says, what the result is: The certificate won't be verified.

Since there are more elegant ways to work with self-signed certificates than performing no verification at all, I'd find `allow_self_signed` rather confusing.

Reply:
If I have to implement such thing, I think that I'll change to way to pass things. Instead of many more parameters given to the constructor, I change the method to that `public function __construct(LoggerInterface $logger = null, array $options = [])`. Then I check if the keys `verify_peer` or `allow_self_signed` are used in the `$options` array and I configure the stream in consequence. But what happen if `verify_peer` and `allow_self_signed` are both passed ?

Reply:
Well you're right, if you had to implement it, it would be better to change the way you pass stream options to the constructor.
Also, I agree with @derrabus, the `allow_self_signed` option is quite confusing and the `verify_peer` and `verify_peer_names` seem to suffice (at least in my current use cases which make use of self signed certificates) so the PR in its current state seems fine enough.